### PR TITLE
move get_grub_file() from libvirt_vm to utils_misc

### PR DESCRIPTION
**Pull request by chloerh**

### CODE REVIEW

**Comment by chloerh** [comment]
@will-Do updated, please review, thanks

File: virttest/utils_misc.py [hunk 1]
```diff
@@ -3469,6 +3469,34 @@ def check_device_driver(pci_id, driver_type):
     return driver == driver_type
 
 
+def get_grub_file(session=None):
```
Comment (xutian) [inline virttest/utils_misc.py]: get_bootloader_cfg looks better, it looks bootloader in power platfrom is not grub.

File: virttest/utils_misc.py [hunk 2]
```diff
@@ -3469,6 +3469,34 @@ def check_device_driver(pci_id, driver_type):
...
+    """
+    grub_paths = [
+        '/boot/grub/grub.conf',
+        '/boot/grub2/grub.cfg',
+        '/etc/grub.conf',
+        '/etc/grub2.cfg',
+    ]
+    grub_path = ''
+    for path in grub_paths:
+        cmd = "ls %s" % path
```
Comment (xutian) [inline virttest/utils_misc.py]: use "test -f %s " % path looks easy to understand.

File: virttest/utils_misc.py [hunk 3]
```diff
@@ -3469,6 +3469,34 @@ def check_device_driver(pci_id, driver_type):
...
 
 
+def get_grub_file(session=None):
+    """
+    Find grub file path of guest or host.
+
+    :param session: session of the vm whose grub path needs to be found.
+    :return: grub file path, empty string if no grub file found.
+    """
+    grub_paths = [
```
Comment (xutian) [inline virttest/utils_misc.py]: bootloader_cfg = [ '/boot/grub/grub.conf',  '/boot/grub/grub.conf',  '/etc/grub.conf', '/etc/grub2.cfg', '/boot/etc/yaboot.conf']

**Comment by chloerh** [comment]
@xutian updated. please review

File: virttest/utils_misc.py [hunk 4]
```diff
@@ -3469,6 +3469,35 @@ def check_device_driver(pci_id, driver_type):
...
+    """
+    bootloader_cfg = [
+        '/boot/grub/grub.conf',
+        '/boot/grub2/grub.cfg',
+        '/etc/grub.conf',
+        '/etc/grub2.cfg',
+        '/boot/etc/yaboot.conf'
+    ]
+    cfg_path = ''
+    for path in bootloader_cfg:
```
Comment (xutian) [inline virttest/utils_misc.py]: how about update your code like below :)

test_func =  session and session.cmd_status or process.system
try:
     return [ cfg for  cfg in bootloader_cfg if not test_func("test -f %s" %  cfg)][-1]
except IndexError:
     logging.error("Failed to locate grub config file in %s." % bootloader_cfg)

Comment (chloerh) [inline virttest/utils_misc.py]: It's a little bit hard to understand, I mean not that straightforward. Could I just keep the way it is?

**Comment by xutian** [comment]
looks ok to me

File: virttest/utils_misc.py [hunk 5]
```diff
@@ -3469,6 +3469,35 @@ def check_device_driver(pci_id, driver_type):
...
+        cmd = "test -f %s" % path
+        if session:
+            status = session.cmd_status(cmd)
+        else:
+            status = process.system(cmd)
+        if not status:
+            cfg_path = path
+    if not cfg_path:
+        logging.error("Failed to locate grub config file "
```
Comment (PandaWei) [inline virttest/utils_misc.py]: You choose 'get_bootloader_cfg' as function name,
It's better to output "Failed to locate bootloader config file".

Comment (chloerh) [inline virttest/utils_misc.py]: Thanks. Updated

**Comment by PandaWei** [comment]
ACK

**Comment by will-Do** [comment]
Thx, let's merge this patch

**Pull request merged**